### PR TITLE
logging function at root level

### PR DESCRIPTION
**Pull request by ghost**
to consolidate logging, we can use parent-child logger functionality
I wrote a `get_logger()` function that simplifies the process of creating a logger with arbitrary parent or logfile

### CODE REVIEW

**Comment by ghost** [comment]
should I make a pull request into [`devel`](https://github.com/jreniel/adcircpy/tree/devel) instead?

**Comment by jreniel** [comment]
Cool, I have the logger on the root level on my other packages, let me look at this for a bit and we can merge into master directly.

File: adcircpy/__init__.py [hunk 1]
```diff
@@ -22,5 +25,71 @@
...
+
+def repository_root(path: str = None) -> str:
+    if path is None:
+        path = __file__
+    if os.path.isfile(path):
+        path = os.path.dirname(path)
+    if '.git' in os.listdir(path):
+        return path
+    else:
+        return repository_root(os.path.dirname(path))
```
Comment (ghost) [inline adcircpy/__init__.py]: added this function to help examples and such find data between directories in the repo

File: adcircpy/__init__.py [hunk 2]
```diff
@@ -22,5 +25,71 @@
...
+                            return rec.levelno in (logging.DEBUG, logging.INFO)
+
+                    console_output = logging.StreamHandler(sys.stdout)
+                    console_output.setLevel(console_level)
+                    console_output.addFilter(LoggingOutputFilter())
+                    logger.addHandler(console_output)
+
+                console_errors = logging.StreamHandler(sys.stderr)
+                console_errors.setLevel(max((console_level, logging.WARNING)))
+                logger.addHandler(console_errors)
```
Comment (ghost) [inline adcircpy/__init__.py]: this splits the console output into stdout and stderr, which helps when looking for errors and warnings. In PyCharm stderr shows up as red, I dunno if default python output on the terminal can be colored though

Comment (jreniel) [reply adcircpy/__init__.py]: put this on a logger.py file

File: adcircpy/logger.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    """
+    Retrieve an instance of `logging.Logger` with split console output (`stdout` and `stderr`).
+
+    :param name: name of logger; use `.` to inherit from parent
+    :param log_filename: path to log file
+    :param file_level: logging level of file output
+    :param console_level: logging level of console output
+    :param name_field_length: number of characters to which to pad the name field
+    :return: instance of `logging.Logger`
+    """
```
Comment (ghost) [inline adcircpy/logger.py:22]: added docstring, removed `log_format` parameter to make it simpler

**Pull request merged**